### ISSUE TITLE
[NuGet Add Using] "Add using" considers a spelling mistake a better match than a case change if the namespace is already imported

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#8685]
If the namespace that contains a type is already imported, Add Using considers a spelling mistake to be a better match than just flipping the case.

Install these packages:

```
<package id="DocumentFormat.OpenXml" version="2.5" targetFramework="net46" />
<package id="EntityFramework" version="6.1.3" targetFramework="net46" />
```

``` C#
using OfficeOpenXml.FormulaParsing.Excel.Functions.Math;

class Program
{
    static void Main(string[] args)
    {
        fact
    }
}

```

Right-click and resolve:

Expected: 
![image](https://cloud.githubusercontent.com/assets/1103906/13031272/77c9ff56-d27c-11e5-8524-79ba65e54e2b.png)

Actual:
![image](https://cloud.githubusercontent.com/assets/1103906/13031273/8a43f042-d27c-11e5-92d5-3bab19bb77d0.png)

If you remove the using, then the correct behavior occurs (Fact is imported)

**Comment by davkean** [comment]
tag @CyrusNajmabadi

**Comment by davkean** [comment]
Here's another repro that I ran into:

``` C#
    class Program
    {
        static void Main(string[] args)
        {
            JToken one;
            jtoken two;
        }
    }
```

Resolve the first one to the JSON.NET NuGet package. Attempt to resolve the second one, and it wants to install a completely unrelated package that contains a type called "Token".

**Comment by michaelcfanning** [comment]
nice bug. :)

**Comment by CyrusNajmabadi** [comment]
The first case (with 'fact') no longer reproes for me.

The second case (with 'jtoken') doesn't repro either.  But hte experience isn't great.  Basically we don't try to fix up the spelling of jtoken.  This would be fixed if we enabled the C# spell checker.  Something i hope to do for Update3.

**Comment by CyrusNajmabadi** [comment]
So i'm not sure what to do with this issue.

**Comment by michaelcfanning** [comment]
is it possible you are done? if you look at the jtoken example, after adding the package ref and typing 'jtoken', we see an expected result: intellisense suggests a correction to JToken. that's the expected path.

why aren't we providing the fix ui in this case? is it because you run the query, determine the match, then detect that the matched package is already added?

**Comment by CyrusNajmabadi** [comment]
> why aren't we providing the fix ui in this case? is it because you run the query, determine the match, then detect that the matched package is already added?

Yes.  

In general, fixing up the case of a reference is something that would be handled by another feature we've got.  Namely the "Spellchecker".  However, we haven't shipped that feature for C# yet, only for VB.  I did the work in Update2 to pull out a common core so that we could have a C# implementation,  Unfortunately, there wasn't enough time to actually provide the C# impl and get it properly tested.  I'm hoping to have that feature implemented in our next release.
